ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -287,18 +287,26 @@ - (void)testCurrentProcessWasSpawnedByRegularUser {
     XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
     MockProcess_Reset();
 
-    // Process is a service user and does not have a parent
+    // Process is not a 'amfid' service user and does not have a parent
     MockProcess_AddContext(context, 500 /*pid*/);
-    MockProcess_SetSelfPid(500);
+    MockProcess_SetSelfInfo(500, "Test");
     MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
     MockProcess_AddProcess(500 /*pid*/, 1 /*credentialId*/, 501 /*ppid*/, "test" /*name*/);
     XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
     MockProcess_Reset();
 
+    // Process is a 'amfid' service user and does not have a parent
+    MockProcess_AddContext(context, 500 /*pid*/);
+    MockProcess_SetSelfInfo(500, "amfid");
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(500 /*pid*/, 1 /*credentialId*/, 501 /*ppid*/, "test" /*name*/);
+    XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
```

Comment:
`CurrentProcessWasSpawnedByRegularUser` is no longer the best name.

I could argue we only have 1 exclusion so keep the name, or `CurrentProcessWasSpawnedByAllowedUser `or similar?  @wilbaker @pmj

Reply:
I like the idea of renaming the function but I am having trouble with name suggestions 😄 

What I'm not a big fan of with `CurrentProcessWasSpawnedByAllowedUser` is that is sounds like we're making the decision based on a user's identity alone.

Perhaps `CurrentProcessWasSpawnedByRegularUserOrCanAlwaysHydrate`? But I know it's a real mouthful.

I suppose another option is to put this new check in its own function `CurrentProcessCanAlwaysHydrate`, but then we have the trouble of always remembering to call both.

I'm also okay with leaving this if we can't think of a better name

Reply:
@wilbaker I'm LOL here cause that was my exact thought process!  

I'm leaning it towards leaving it as is for now.  If we add a second exclusion in the future, I think we definitely need to rename and/or split it out.

Reply:
One other option, if we name it based on what question it's trying to answer: `CurrentProcessIsAllowedToHydrate`, or if we flipped the return value `CurrentProcessIsRestrictedFromHydrating`

Reply:
I went with `CurrentProcessIsAllowedToHydrate ` for clarity.  Thanks for the opinions here @wilbaker!